assets/js/pages/SettingsPage/SettingsPage.jsx

```diff
@@ -24,28 +25,52 @@ import {
   getSoftwareUpdatesSettingsErrors,
 } from '@state/selectors/softwareUpdatesSettings';
 
+import { dismissNotification } from '@state/notifications';
+import { API_KEY_EXPIRATION_NOTIFICATION_ID } from '@state/sagas/settings';
+
 import SuseManagerConfig from '@common/SuseManagerConfig';
+import { format, parseISO } from 'date-fns';
 
 function SettingsPage() {
   const dispatch = useDispatch();
 
   const [loading, setLoading] = useState(false);
   const [apiKey, setApiKey] = useState(null);
-  const [showApiKey, setShowApiKey] = useState(false);
+  const [apiKeyExpiration, setApiKeyExpiration] = useState(null);
+  const [apiKeySettingModalOpen, setApiKeySettingsModalOpen] = useState(false);
   const [clearingSoftwareUpdatesSettings, setClearingSoftwareUpdatesSettings] =
     useState(false);
 
-  useEffect(() => {
-    setLoading(true);
-    get('/installation/api-key')
-      .then(({ data: { api_key: newApiKey } }) => {
-        apiKey !== undefined && setApiKey(newApiKey);
-        setLoading(false);
-      })
+  const fetchApiKeySettings = () =>
+    get('/settings/api_key')
+      .then(
+        ({ data: { generated_api_key: newApiKey, expire_at: expireAt } }) => {
+          setApiKey(newApiKey);
+          setApiKeyExpiration(expireAt);
+          setLoading(false);
+        }
+      )
```

Comment:
Wouldn't be better to include a `finally` clause here, after the `catch` where we don `setLoading(false)`?
This one, we just do it once.
The same for the `saveApiKeySettings`, that if the `patch` fails, we don't set the loading value otherwise

Reply:
done